tests/backend_api/test_connector.py

```diff
@@ -123,23 +147,32 @@ def test_fetch_merged_items_mocked(
     )
 
 
-def test_get_merged_item_mocked(
+def test_fetch_all_merged_items_mocked(
     mocked_backend: MagicMock, merged_person: MergedPerson
 ) -> None:
-    mocked_return = {"items": [merged_person.model_dump()], "total": 1}
+    mocked_return = {"items": [merged_person.model_dump()], "total": 103}
     mocked_backend.return_value.json.return_value = mocked_return
 
     connector = BackendApiConnector.get()
-    response = connector.get_merged_item("NGwfzG8ROsrvIiQIVDVy")
+    items = list(
+        connector.fetch_all_merged_items(
+            query_string="Tintzmann",
+        )
+    )
 
-    assert response == merged_person
+    assert items == [merged_person] * 2  # should be two requests, one Tintzmann each
```

Comment:
It took me some time to understand why we expect two Tintzmanns. A more detailed comment probably helps future code readers.

```suggestion
    assert items == [merged_person] * 2  # two requests to mocked backend due to "total" > "item_number_limit", each mocked request returns one Tintzmann
```

Reply:
cleaned it up https://github.com/robert-koch-institut/mex-common/pull/497/commits/f6dfa796f46b2c3ac3bff9d0e758a5878abf64d7

Reply:
even better, thank you!